src/otc_sccp.erl

```diff
@@ -330,32 +339,31 @@ separate_fields_test() ->
     CgPA = <<4, "CgPA">>,
     D = <<1, "D">>,
     Bin = <<CgPA/binary, D/binary, CdPA/binary>>,
-    Vs = separate_fields(<<94, 3, 14, 0>>, Bin),
-    ?assertEqual([<<"CdPA">>, <<"CgPA">>, <<"D">>, <<>>], Vs).
+    ?assertEqual([<<"CgPA">>,<<"CgPA">>,<<"CgPA">>,<<"D">>], separate_fields(<<4, 3, 2, 6>>, Bin)),
+    ?assertEqual([<<"CdPA">>,<<"D">>,<<"CgPA">>,<<>>], separate_fields(<<11, 8, 2, 0>>, Bin, 4)).
```

Comment:
not sure I understand these tests.
There is an expected order: `[CdPA, CgPA, LD, OptBin]`
a binary: `<<CgPABin1, LDBin2, CdPABin3, OptBin4>>`
and a pointer binary: `<<Last, Second, Third, First>>`

why would it return `CgPA, CgPA, CgPA, D` or unsorted `CdPA, D, CgPA, Opt`?

looking at the code, the separate_fields need to return the arguments in sorted order. so it would make sense to test that.
e.g. [FirstArg, SecondArg, ThirdArg]
`[CdPA, CgPA, LD] = separate_fields(<<11, 2, 8, 0>>, Bin, 4).`

Reply:
I think these tests are just used as validation of that it is able to pick things with weird pointer not that the message makes sense.

Reply:
Updated the test to describe better what they do. Pointers can be received in arbitrary order.